token.go

```diff
@@ -239,7 +239,7 @@ type tokenRing struct {
 	hosts       []*HostInfo
 }
 
-func newTokenRing(partitioner string, hosts []*HostInfo) (*tokenRing, error) {
+func newTokenRing(partitioner string, hosts []HostInfo) (*tokenRing, error) {
```

Comment:
Originally I used ```[]*HostInfo``` but I had to change the ringDescriber to provide ```[]*HostInfo``` since it provides ```[]HostInfo```. However, I decided to avoid changing the ringDescriber or the SetHosts function in ```ConnectionPool``` and do it this way. In its favor, ```[]HostInfo``` does have the benefit of being more compact in memory.